py-polars/tests/unit/datatypes/test_struct.py

```diff
@@ -1131,3 +1131,21 @@ def test_zfs_row_encoding(size: int) -> None:
 
     # We need to ignore the order because the group_by is non-deterministic
     assert_frame_equal(gb, df, check_row_order=False)
+
+
+@pytest.mark.may_fail_auto_streaming
+def test_list_to_struct_19208() -> None:
+    df = pl.DataFrame(
+        {
+            "nested": [
+                [{"a": 1}],
+                [],
+                [{"a": 3}],
+            ]
+        }
+    )
+    assert pl.concat([df[0], df[1], df[2]]).select(
+        pl.col("nested").list.to_struct()
```

Comment:
@orlp this expression's dtype is dependent on the data and therefore not suitable for streaming. I want to add a flag for those expressions so that they will fallback to in-memory.